domain/removal/service/secret.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	coreapplication "github.com/juju/juju/core/application"
+	coreunit "github.com/juju/juju/core/unit"
+	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/secrets/provider"
+	"github.com/juju/juju/internal/secrets/provider/juju"
+)
+
+type SecretModelState interface {
+	// DeleteApplicationOwnedSecretContent deletes content for all
+	// secrets owned by the application with the input UUID.
+	// It must only be called in the context of application removal.
+	DeleteApplicationOwnedSecretContent(ctx context.Context, aUUID string) error
+
+	// DeleteUnitOwnedSecretContent deletes content for all
+	// secrets owned by the unit with the input UUID.
+	// It must only be called in the context of unit removal.
+	DeleteUnitOwnedSecretContent(ctx context.Context, uUUID string) error
+}
+
+func (s *Service) deleteApplicationOwnedSecrets(ctx context.Context, aUUID coreapplication.UUID) error {
```

Comment:
This removes the content only - not the metadata. Can we update the method name to reflect that?
"Secrets" as a term implies that the whole shebang will be removed - metadata and content, when that's not the case.

Reply:
It will delete everything. I'll add the other parts in following patches.